src/tbb/governor.h

```diff
@@ -148,6 +155,69 @@ class governor {
         return false;
 #endif
     }
+
+#if __TBB_STATISTICS
+    enum class statistics : std::size_t {
+        executed = 0,
+        try_get,
+        got,
+        try_steal,
+        stolen,
+        empty_vic_pool,
+        size,
+    };
+    using array_stat = std::array<std::uint32_t, static_cast<std::size_t>(statistics::size)>;
+
+    class collect_statistics : no_copy {
+    public:
+        collect_statistics() = default;
+        ~collect_statistics() {
+            print_statistics();
+        }
+        void emplace(array_stat thread_stat) {
+            std::lock_guard<std::mutex> lk {m_};
+            accumulate_statistics.emplace_back(thread_stat);
+        }
+
+        // thread-unsafe method
+        void print_statistics() {
+            int offset = 2;
+            /* std::string not so bad because of Small String Optimization */
+            const static std::string column_names[] = {
+                "executed", "try get",
+                "got", "try steal", "stolen",
+                "empty vic pool",
+            };
+            for (auto& el : column_names) {
+                std::cout << std::left << std::setw(el.size() + offset) << el;
```

Comment:
This is not very convenient, that we should remember order in this array, to solve this problem we can add a part of magic enum in utils, maybe : https://github.com/Neargye/magic_enum - its statically convert enum into string using `__PRETTY_FUNCTION__` compilers extension

Reply:
I think that using a third-party solution is overcomplicated for this task, what do you think about storing the strings inside the map where the keys are enum values and the values are strings?

Reply:
I think that using a third-party component is an overcomplicated solution for this purpose. What do you think about storing the strings inside a map where the keys are the corresponding enum values?